core/src/main/java/org/mapstruct/Javadoc.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.CLASS)
+public @interface Javadoc {
```

Comment:
Can we please add some javadoc with examples for this? Please also add javadoc for all the other methods.

Reply:
Of course I have provided an initial implementation. I'm sorry, I should have included the necessary comments from the beginning.